src/metrics/encoding/protobuf/aggregated_decoder.go

```diff
@@ -62,7 +62,8 @@ func (d AggregatedDecoder) Value() float64 {
 	return d.pb.Metric.TimedMetric.Value
 }
 
-// Annotation returns the decoded annotation.
+// Annotation returns the decoded annotation. Note that it might get invalidated when
+// decoders are pooled and thus should be copied on access.
```

Comment:
I'm not 100% sure about the validity of this claim, and it looks like it is not related to the change in this PR, so maybe let's defer it until another PR?

Reply:
https://github.com/m3db/m3/pull/3605/commits/0e5a0e7076764cb1253b4822bc1ee20fb0ef5350